actors/common/src/main/java/org/sunbird/learner/util/Util.java

```diff
@@ -600,6 +601,12 @@ public static SearchDTO createSearchDto(Map<String, Object> searchQueryMap) {
         search.setLimit(((BigInteger) searchQueryMap.get(JsonKey.LIMIT)).intValue());
       }
     }
+    if (search.getLimit() > DEFAULT_ELASTIC_DATA_LIMIT) {
+      search.setLimit(DEFAULT_ELASTIC_DATA_LIMIT);
+    }
+    if (search.getLimit() + search.getOffset() > DEFAULT_ELASTIC_DATA_LIMIT) {
+      search.setOffset(DEFAULT_ELASTIC_DATA_LIMIT - search.getLimit());
```

Comment:
instead of search.setOffset(DEFAULT_ELASTIC_DATA_LIMIT - search.getLimit());
  we need to reduce limit size other wise it will have come duplicate records at end.